www/cherokee/Portfile

```diff
@@ -1,80 +1,101 @@
 PortSystem      1.0
 PortGroup       old_openssl 1.0
+PortGroup       github      1.0
 
+github.setup    cherokee webserver 1.2.104 v
 name            cherokee
-version         1.2.101
-revision        2
-set branch      [join [lrange [split ${version} .] 0 1] .]
+revision        0
+
 categories      www
 platforms       darwin
 maintainers     nomaintainer
 license         GPL-2
 
+homepage        https://www.cherokee-project.com/
+
 description     Cherokee web server
+
 long_description \
   Cherokee is a relatively new HTTP server designed to be as\
   performant as Apache and lighttpd, but much easier to \
   configure and maintain.  It ships out-of-the-box SCGI \
   support, making it ideal for deploying dynamic websites.
-homepage        http://www.cherokee-project.com/
-master_sites    ${homepage}download/${branch}/${version}
 
-checksums       md5     ef47003355a2e368e4d9596cd070ef23 \
-                sha1    b27f149c7d7111207ac8c3cd8a4856c05490d136 \
-                rmd160  dd3dedc352ba17bdcefd8e200143b8ffa19ad035
+checksums       rmd160  f5271b4b6f6be3cf70a88a3933fd90597862b289 \
+                sha256  5cbd00ff48503eaf90356b2975e311c02977f9166927e57fc23f541a109efd98 \
+                size    5312997
 
-depends_lib     port:pcre \
-                port:zlib \
-                port:libgeoip \
-                port:rrdtool
+set ck_wwwroot  ${prefix}/var/www/htdocs
+set ck_cgibin   ${prefix}/var/www/cgi-bin
+set ck_lcl_bin  ${workpath}/bin
 
-openssl.branch 1.0
-openssl.configure build_flags
+post-extract {
+    reinplace -W ${worksrcpath} \
+        "s|/Library/WebServer/Documents|${ck_wwwroot}|g" \
+        admin/util.py configure.ac
 
-# Startup item.
-set cherokee_config_name   cherokee.conf
-set cherokee_config        ${prefix}/etc/${name}/${cherokee_config_name}
-set cherokee_pidfile_name  cherokee.pid
-set cherokee_pidfile       ${prefix}/var/run/${cherokee_pidfile_name}
-
-startupitem.create  yes
-startupitem.init    "PIDFILE=${cherokee_pidfile}"
-startupitem.start   "${prefix}/sbin/cherokee -C ${cherokee_config}"
-startupitem.stop    "kill \$(cat \$PIDFILE)"
-
-variant no_startupitem description {Do not create a startup item} {
-    startupitem.create  no
-}
+    reinplace -W ${worksrcpath} \
+        "s|/Library/WebServer/CGI-Executables|${ck_cgibin}|g" configure.ac
 
-variant no_ipv6 description {Disable IPv6 support} {
-    configure.args-append --disable-ipv6
+    file mkdir ${ck_lcl_bin}
+    # cherokee's configure is specifically looking for a python2 symlink.
+    file link -symbolic ${ck_lcl_bin}/python2 ${prefix}/bin/python2.7
```

Comment:
Note that MacPorts has an `ln` procedure that works like the `ln` executable so you could just write:
```
ln -s ${prefix}/bin/python2.7 ${ck_lcl_bin}/python2
```

Reply:
Noted, changed